### DISCUSSION TITLE
Circular route

### DISCUSSION THREAD

**Discussion by ramondevic** [devemux86/cruiser#430] [Q&A] [answered]
Can't you make a circular route with BRouter?
With graphHopper yes it is possible.

I wanted to do an offroad route with BRouter by being able to assign/avoid different types of roads.

BRouter
![Screenshot_20240126-074055](https://github.com/devemux86/cruiser/assets/122691401/f0a42786-097a-4f4a-ad39-f4ccba9e4ca2)

GraphHopper
![Screenshot_20240126-074013](https://github.com/devemux86/cruiser/assets/122691401/fa18f6ac-9fae-4b59-975d-f796fd88df49) ![Screenshot_20240126-074143](https://github.com/devemux86/cruiser/assets/122691401/1387900d-3fb3-4e0c-8fd0-659559bce121)

**Comment by devemux86** [comment, marked as answer]
@ramondevic

There is no automatic round trip generation, it will be provided in the future.

Round trips are possible with all routers by setting manually extra waypoints.
It's better because we can define exactly where we want the round trip to be.

**Reply by ramondevic** [reply]
In Brouter there is not even the possibility to do the circular route and with GraphHopper yes, I don't understand it.
Have you tried it?

**Reply by devemux86** [reply]
@ramondevic 

We can do round trips with all the routers.
You can set waypoints manually in a circle (or other shape) with the distance and direction you want.

Automatic round trips just set random waypoints and don't respect the distance / direction we choose.
So we always have to correct them afterwards.

**Reply by ramondevic** [reply]
No, the problem I have is that with BRouter when clicking on a favorite (or at any point on the map) this window opens and there is no possibility to choose "circular route"
![299907145-f0a42786-097a-4f4a-ad39-f4ccba9e4ca2](https://github.com/devemux86/cruiser/assets/122691401/e3e329cf-517b-42f6-a24d-62a8972fca11)

What am I doing wrong?

**Reply by ramondevic** [reply]
It is clear in the first post.

**Reply by devemux86** [reply]
@ramondevic there is no automatic way to create circular routes.

We can manually set waypoints and create routes of any shape.

**Reply by 0709wiwiwi** [reply]
@ramondevic 
Simple, there is no such a BRouter offer yet.
https://github.com/abrensch/brouter/issues/460
https://github.com/nrenner/brouter-web/wiki/Round-trip-suggestions

**Reply by ramondevic** [reply]
Crec que no se m'entén o no m'explico prou bé.

Algú ha provat si funciona bé o no ?

No sé que faig malament.
Cruiser 3.0.38-b13

**Reply by devemux86** [reply]
@ramondevic no hi ha cap manera automàtica de crear rutes circulars.

Podem establir waypoints manualment i crear rutes de qualsevol forma.

**Reply by ramondevic** [reply]
Oh thanks !

devemux86 says it works and I tried a lot.

So the anser is IT DOS NOT WORKS WITH BROUTER.
